backend/packages/Upgrade/src/api/services/ExperimentService.ts

```diff
@@ -542,7 +533,7 @@ export class ExperimentService {
         if (promiseArray.length) {
           const metricsDocs = await Promise.all([...promiseArray]);
           queriesDocToSave = queriesDocToSave.map((queryDoc, index) => {
-            queryDoc.metric = metricsDocs[index];
+            metricsDocs ? queryDoc.metric = metricsDocs[index]: queryDoc.metric = null;
```

Comment:
You should not save the query with no valid metric.

Reply:
okay